composer/utils/checkpoint.py

```diff
@@ -532,32 +531,55 @@ def save_checkpoint(
 
     # Sharded checkpoints get their own little folder.
     if state.fsdp_sharded_state_dict_enabled:
+        if using_torch_2():
+            # Make state dict.
+            if weights_only:
+                state_dict = {'model': state.state_dict()['model']}
```

Comment:
```suggestion
            if weights_only:
                state_dict['model'] = state_dict['state']['model']
                del state_dict['state']
```
Weights only should keep rng per logic above?

Reply:
ended up keeping the old logic except for when weights_only is False and torch >=2.0